crates/ruff_linter/src/rules/airflow/rules/removal_in_3.rs

```diff
@@ -640,12 +651,57 @@ fn removed_name(checker: &mut Checker, expr: &Expr, ranged: impl Ranged) {
             Airflow3Removal {
                 deprecated,
                 replacement,
+                is_context_variable: false,
             },
             ranged.range(),
         ));
     }
 }
 
+fn extract_name_from_slice(slice: &Expr) -> Option<String> {
+    if let Expr::StringLiteral(ExprStringLiteral { value, .. }) = slice {
+        Some(value.to_string())
+    } else {
+        None
+    }
+}
+
+pub(crate) fn removed_context_variable(checker: &mut Checker, expr: &Expr) {
+    if let Expr::Subscript(ExprSubscript { value, slice, .. }) = expr {
+        if let Expr::Name(ExprName { id, .. }) = &**value {
+            if id.as_str() == "context" {
+                if let Some(key) = extract_name_from_slice(slice) {
+                    const REMOVED_CONTEXT_KEYS: [&str; 11] = [
+                        "execution_date",
```

Comment:
"conf" had also been removed

Reply:
It's tracked in https://github.com/apache/airflow/issues/45212, but yep, we could do it here as well

Reply:
If we want to add `conf` here, it would be awesome if we could include `triggering_dataset_events` → `triggering_asset_events`. If not, I can make a separate PR

Reply:
I have added `conf` removal check. i`nclude triggering_dataset_events` → `triggering_asset_events` can be part of another PR